lib/normalizer.c

```diff
@@ -1270,6 +1278,14 @@ grn_nfkc_normalize_unify_stateless(grn_ctx *ctx,
       char_type = data->options->char_type_func(current);
     }
 
+    if (before && data->options->unify_alphabet_diacritical_mark &&
+        GRN_CHAR_TYPE(char_type) == GRN_CHAR_ALPHA &&
+        unified_char_length == 2) {
+      unifying =
+        grn_nfkc_normalize_unify_alphabet_diacritical_mark(unifying,
+                                                           unified_alpha);
```

Comment:
We also need to update `unified_char_length` when we we unify `unifying` because `unified_alpha` is 1 byte character.
```suggestion
      unifying =
        grn_nfkc_normalize_unify_alphabet_diacritical_mark(unifying,
                                                           unified_alpha);
      if (unifying == unified_alpha) {
        unified_char_length = sizeof(unified_alpha);
      }
```

Reply:
fix: [9d05eed](https://github.com/groonga/groonga/pull/1831/commits/9d05eed1511ea76d7ca31da08a856bce0ca166c9) Thanks.
📝 If the character isn't unified, we should keep the size of characters 2 bytes.